source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -178,29 +162,12 @@ ShaderPtr OslShaderGenerator::generate(const string& name, ElementPtr element, G
         }
     }
 
-    // Emit final outputs
-    if (isSurfaceShaderOutput)
-    {
-        // Special case for having 'surfaceshader' as final output type.
-        // This type is a struct internally (BSDF, EDF, opacity) so we must
-        // convert this to a single closure color type in order for renderers
-        // to understand this output.
-        const ShaderGraphOutputSocket* socket = graph.getOutputSocket(0);
-        const string result = getUpstreamResult(socket, context);
-        emitScopeBegin(stage);
-        emitLine("float opacity_weight = clamp(" + result + ".opacity, 0.0, 1.0)", stage);
-        emitLine(singleOutput->getVariable() + " = (" + result + ".bsdf + " + result + ".edf) * opacity_weight + transparent() * (1.0 - opacity_weight)", stage);
-        emitScopeEnd(stage);
-    }
-    else
+    // Assign results to final outputs.
+    for (size_t i = 0; i < outputs.size(); ++i)
     {
-        // Assign results to final outputs.
-        for (size_t i = 0; i < outputs.size(); ++i)
-        {
-            const ShaderGraphOutputSocket* outputSocket = graph.getOutputSocket(i);
-            const string result = getUpstreamResult(outputSocket, context);
-            emitLine(outputSocket->getVariable() + " = " + result, stage);
-        }
+        const ShaderGraphOutputSocket* outputSocket = graph.getOutputSocket(i);
+        const string result = getUpstreamResult(outputSocket, context);
```

Comment:
Minor note: Was this intended to be a `const` reference to a string?  If so, then we should add the reference symbol; otherwise, this would be clearer as a non-const string value.